src/Utilities/Math.hpp

```diff
@@ -79,3 +79,24 @@ template <typename T, Requires<std::is_arithmetic<T>::value> = nullptr>
 auto invcbrt(const T& arg) noexcept {
   return static_cast<T>(1.0) / cbrt(arg);
 }
+
+namespace sgn_detail {
+template <typename T>
+constexpr int sgn(const T& val, std::true_type /*is_signed*/) noexcept {
+  return static_cast<int>(static_cast<bool>(static_cast<T>(0) < val)) -
+         static_cast<int>(static_cast<bool>(val < static_cast<T>(0)));
+}
+
+template <typename T>
+constexpr int sgn(const T& val, std::false_type /*is_signed*/) noexcept {
+  return static_cast<int>(static_cast<bool>(static_cast<T>(0) < val));
+}
+}  // namespace sgn_detail
+
+/// \ingroup UtilitiesGroup
+/// \brief Compute the sign function of `val` defined as `1` if `val > 0`, `0`
+/// if `val == 0`, and `-1` if `val < 0`.
+template <typename T>
+constexpr int sgn(const T& val) noexcept {
```

Comment:
Is it deliberate to return int? At very least, this seems counter-intuitive to me for floating point types (e.g. our step function returns floating points).
From the definition of the sign function, unsigned types should also never get a negative value, so I don't see any immediate danger to just returning `T`. Might need yet another cast, though :P .

Reply:
Yes it's intentional, but I agree there's no harm in using `T`, I'll change it.